### DISCUSSION TITLE
How to handle desaster recovery startup order?

### DISCUSSION THREAD

**Discussion by ReTaec** [quartznet/quartznet#2531] [Q&A]
Hey, 
i am working on a microservice which is using quartz.net with a persistent store on an SQL server. Every morning (after the startup of the environments) the service trows a 500.30 Error and the service is terminated. 
After some deeper look on it, i noticed a "Quartz.SchedulerException: Database schema validation failed" Error in the logs - which makes sense, becaue the service is starting faster than the SQL Server instance (running on the same machine) so it can not validate the schema. When I restart the service after SQL server is running, the service runs without any issue and as expected. 

My question is: is there any redo logic available for this schema validation? In production this behavoiur should (in best case) never happen, but it would be nice for the dev / test environments

**Comment by lahma** [comment]
Maybe easiest to add startup delay for scheduler in quartz options.

https://github.com/quartznet/quartznet/blob/fc7d13393068205adec6f802b2a86c9dbcd0eace/src/Quartz.Examples.Worker/Program.cs#L94

**Reply by ReTaec** [reply]
thnaks for the quick reply and idea! but wouldn't this result in a delay even if the database is present?

**Reply by lahma** [reply]
Well it's just development environment like you said.

**Reply by ReTaec** [reply]
true thing :) but i can hear our dev's crying after every change "bohoo its taking so long" :D Ok so in summary i got the following: no direct retry logic, but a delay could be an option (with possible delay on normal starups)
thank you very much for your help and ideas!

**Reply by ReTaec** [reply]
is it possible to create some kind of custom startup code, which could handle the DB check? something like `options.StartScript = SomeAdditionalClass();`

**Reply by lahma** [reply]
Maybe just disable the check if that's the problem. Generally in container environments I'd use checks that check if containers/pods are ready to start based on their dependencies. If you want custom logic you could create custom hosted service which would ping your infra before starting the scheduler.
